sim/rogue/rogue.go

```diff
@@ -174,62 +185,62 @@ func (rogue *Rogue) Reset(sim *core.Simulation) {
+	var secondaryModifier float64
+	if applyLethality {
+		secondaryModifier += 0.1 * float64(rogue.Talents.Lethality)
+	}
+	return rogue.Character.MeleeCritMultiplier(1.0, secondaryModifier)
+}
+func (rogue *Rogue) SpellCritMultiplier() float64 {
+	return rogue.Character.SpellCritMultiplier(1, 0)
+}
+
+func NewRogue(character *core.Character, options *proto.RogueOptions, talents string) *Rogue {
 	rogue := &Rogue{
 		Character: *character,
 		Talents:   &proto.RogueTalents{},
+		Options:   options,
 	}
-	// core.FillTalentsProto(rogue.Talents.ProtoReflect(), talents, TalentTreeSizes)
-
-	// // Passive rogue threat reduction: https://wotlk.wowhead.com/spell=21184/rogue-passive-dnd
-	// rogue.PseudoStats.ThreatMultiplier *= 0.71
-	// rogue.PseudoStats.CanParry = true
-
-	// maxEnergy := 100.0
-	// if rogue.Talents.Vigor {
-	// 	maxEnergy += 10
-	// }
-	// if rogue.HasMajorGlyph(proto.RogueMajorGlyph_GlyphOfVigor) {
-	// 	maxEnergy += 10
-	// }
-	// if rogue.HasSetBonus(Arena, 4) {
-	// 	maxEnergy += 10
-	// }
-	// rogue.EnableEnergyBar(maxEnergy)
-	// rogue.ApplyEnergyTickMultiplier([]float64{0, 0.08, 0.16, 0.25}[rogue.Talents.Vitality])
-
-	// rogue.EnableAutoAttacks(rogue, core.AutoAttackOptions{
-	// 	MainHand:       rogue.WeaponFromMainHand(0), // Set crit multiplier later when we have targets.
-	// 	OffHand:        rogue.WeaponFromOffHand(0),  // Set crit multiplier later when we have targets.
-	// 	AutoSwingMelee: true,
-	// })
-	// rogue.applyPoisons()
-
-	// rogue.AddStatDependency(stats.Strength, stats.AttackPower, 1)
-	// rogue.AddStatDependency(stats.Agility, stats.AttackPower, 1)
-	// rogue.AddStatDependency(stats.Agility, stats.MeleeCrit, core.CritPerAgiMaxLevel[character.Class]*core.CritRatingPerCritChance)
+	core.FillTalentsProto(rogue.Talents.ProtoReflect(), talents, TalentTreeSizes)
+
+	// Passive rogue threat reduction: https://wotlk.wowhead.com/spell=21184/rogue-passive-dnd
+	rogue.PseudoStats.ThreatMultiplier *= 0.71
+	rogue.PseudoStats.CanParry = true
+
+	maxEnergy := 100.0
+	if rogue.HasSetBonus(Arena, 4) {
+		maxEnergy += 10
+	}
+	if rogue.Spec == proto.Spec_SpecAssassinationRogue &&
+		rogue.GetMHWeapon().WeaponType == proto.WeaponType_WeaponTypeDagger {
+		maxEnergy += 20
+	}
+	rogue.EnableEnergyBar(maxEnergy)
+
+	rogue.EnableAutoAttacks(rogue, core.AutoAttackOptions{
+		MainHand:       rogue.WeaponFromMainHand(0), // Set crit multiplier later when we have targets.
+		OffHand:        rogue.WeaponFromOffHand(0),  // Set crit multiplier later when we have targets.
+		Ranged:         rogue.WeaponFromRanged(0),
+		AutoSwingMelee: true,
+	})
+	rogue.applyPoisons()
+
+	rogue.AddStatDependency(stats.Strength, stats.AttackPower, 1)
+	rogue.AddStatDependency(stats.Agility, stats.AttackPower, 2)
+	rogue.AddStatDependency(stats.Agility, stats.MeleeCrit, core.CritPerAgiMaxLevel[character.Class]*core.CritRatingPerCritChance)
+	// Make an assumption we're wearing leather for Leather Armor Spec
+	rogue.MultiplyStat(stats.Agility, 1.05)
```

Comment:
I dont like this assumption. We should add this to core instead and have it properly handled and checked for what the characters are wearing